### ISSUE TITLE
Neutral Players color nitpicking

### ISSUE THREAD

**Issue by InsanityAI** [Warcraft-III-Community-Edition/W3CE#40]
So, in 1.29 and earlier versions, Neutrals were turned into Black player color.
(From left-to-right: Top row: Gray, Coal, Black (Red unit colored as Black with RF Editor), Bottom row: Hostile, Victim, Extra (or Killer), Passive)
![image](https://github.com/Warcraft-III-Community-Edition/W3CE/assets/35784470/7d42bd51-acca-4586-83cb-bbfd4152e23f)

Comparing this to current reforged, we can see that this was fixed:

![image](https://github.com/Warcraft-III-Community-Edition/W3CE/assets/35784470/b9b59f96-deb0-44e2-b097-2a0a9d2202c6)

(Additionally, it seems like Coal player's color is not consistent between the versions as well.)

Neutrals seem to share the same unit color. Which may see correct in 1.29, until you look at Reforged.

CE:
![image](https://github.com/Warcraft-III-Community-Edition/W3CE/assets/35784470/5f3037a9-8228-47a6-bc1f-6dd3238ab69f)

Reforged:
![image](https://github.com/Warcraft-III-Community-Edition/W3CE/assets/35784470/a01099e5-b9ec-4c39-a3c8-426e5169494b)

Neutral Hostile, while being completely black on the minimap shares the same unit color as the rest of neutrals, and none of them is that dark shade that's being portrayed in 1.29.2/CE version

Seems like Black color for units is that same black color (according to GUI-2-jass conversion, it's using `call SetUnitColor( gg_unit_hpea_0007, ConvertPlayerColor(24) )` ) (Trigger editor has no distinction between the hostile black and passive black for unit coloring)

Here's the demo map for this issue:

[neutralColors.zip](https://github.com/user-attachments/files/16163974/neutralColors.zip)

**Comment by InsanityAI** [comment]
TLDR:
Hostile player must have black color on minimap, but must share the same color with other neutrals, while other neutrals have some lighter black/darker gray shade going on
